kedro/config/common.py

```diff
@@ -236,29 +286,3 @@ def _check_duplicate_keys(
     if duplicates:
         dup_str = "\n- ".join(duplicates)
         raise ValueError(f"Duplicate keys found in {filepath} and:\n- {dup_str}")
-
-
-def _path_lookup(conf_path: Path, patterns: Iterable[str]) -> Set[Path]:
-    """Return a set of all configuration files from ``conf_path`` or
-    its subdirectories, which satisfy a given list of glob patterns.
-
-    Args:
-        conf_path: Path to configuration directory.
-        patterns: List of glob patterns to match the filenames against.
-
-    Returns:
-        A set of paths to configuration files.
-
-    """
-    config_files = set()
-    conf_path = conf_path.resolve()
-
-    for pattern in patterns:
-        # `Path.glob()` ignores the files if pattern ends with "**",
-        # therefore iglob is used instead
-        for each in iglob(str(conf_path / pattern), recursive=True):
-            path = Path(each).resolve()
-            if path.is_file() and path.suffix in SUPPORTED_EXTENSIONS:
-                config_files.add(path)
-
-    return config_files
```

Comment:
I refactored this into `_lookup_config_filepaths` and `_is_valid_config_path` above.